utils/flare.py

```diff
@@ -279,6 +321,23 @@ def _add_command_output_tar(self, name, command):
         self._tar.add(temp_path, os.path.join(self._prefix, name))
         os.remove(temp_path)
 
+    # Capture the output of a command (from both std streams and loggers) and the
+    # value returned by the command
+    def _capture_output(self, command):
+        backup_out, backup_err = sys.stdout, sys.stderr
+        out, err = StringIO.StringIO(), StringIO.StringIO()
+        backup_handlers = logging.root.handlers[:]
+        logging.root.handlers = [logging.StreamHandler(out)]
+        sys.stdout, sys.stderr = out, err
+        try:
+            return_value = command()
+        finally:
+            # Stop capturing in a `finally` block to let uncaught exceptions be printed out
```

Comment:
In `stdout`,

After discussing it we should probably catch all exceptions and print them in the file that is added to the tar, I'll change that.